src/components/enterprise-page/data/hooks.js

```diff
@@ -109,45 +111,44 @@ export const useEnterpriseCustomerConfig = (enterpriseSlug, useCache = true) =>
  * @param {string} [enterpriseId] enterprise UUID
  * @param {object} [user] user object containing JWT roles
  *
- * Sets the user's active enterprise and forces login_refresh to re-order the roles inside the user's JWT.
+ * Sets the user's active enterprise.
  */
 export const useUpdateActiveEnterpriseForUser = ({
   enterpriseId,
   user,
 }) => {
-  const [isLoading, setIsLoading] = useState(false);
-
-  const updateActiveEnterpriseAndRefreshJWT = useCallback(async () => {
-    setIsLoading(true);
-
-    try {
-      await updateUserActiveEnterprise(enterpriseId);
-      await loginRefresh();
-    } catch (error) {
-      logError(error);
-    } finally {
-      setIsLoading(false);
-    }
-  }, [enterpriseId]);
+  // Sets up POST call to update active enterprise.
+  const { mutate, isLoading: isUpdatingActiveEnterprise } = useMutation({
+    mutationFn: () => updateUserActiveEnterprise(enterpriseId),
+    onError: () => {
+      logError("Failed to update user's active enterprise");
+    },
+  });
+  const { username } = user;
+  const {
+    data,
+    isLoading: isLoadingActiveEnterprise,
+  } = useQuery({
+    queryKey: ['activeLinkedEnterpriseCustomer', username],
+    queryFn: () => getLinkedEnterprises(username),
+    meta: {
+      errorMessage: "Failed to fetch user's active enterprise",
+    },
+  });
 
   useEffect(() => {
-    if (!(enterpriseId && user)) {
+    if (!data || !enterpriseId) { return; }
+    // Ensure that the current enterprise is linked and can be activated for the user
+    if (!data.find(enterprise => enterprise.enterpriseCustomer.uuid === enterpriseId)) {
```

Comment:
[curious/suggestion] Does the API data also return whether the learner is `linked: true`? Wondering if we should also verify `linked` as part of this conditional. When a learner is de-linked from an enterprise customer, their `EnterpriseCustomerUser` record remains, but updated to `linked=False`.

I imagine the POST call to set active customer may fail if the `EnterpriseCustomerUser` record denotes not being linked?

Reply:
so... ensure they're linked to at least 1 enterprise?

Reply:
would they have access to the learner portal if they weren't linked to any enterprise?

Reply:
regardless I think we could probably do a `if (!activeLinkedEnterprise) { return; }`

Reply:
> would they have access to the learner portal if they weren't linked to any enterprise?

Oh, d'oh. You're right... if the user isn't linked, it won't return in the `enterprise-learner` API response regardless. Disregard this comment :D